frontend/src/components/forms/projectFormTabs/GeneralFormTab.tsx

```diff
@@ -1,59 +1,93 @@
 import { DatePicker, Form, FormInstance, Input, Switch, Typography } from "antd"
 import { useTranslation } from "react-i18next"
-import { FC } from "react"
+import { FC, useEffect, useState } from "react"
 import MarkdownEditor from "../../input/MarkdownEditor"
 
 const GeneralFormTab: FC<{ form: FormInstance }> = ({ form }) => {
-  const { t } = useTranslation()
-  const description = Form.useWatch("description", form)
-
-  return (
-    <>
-      <Form.Item
-        label={t("project.change.name")}
-        name="name"
-        rules={[{ required: true, message: t("project.change.nameMessage") }]}
-      >
-        <Input />
-      </Form.Item>
-
-    <Typography.Text>
-        {t("project.change.description")}
-    </Typography.Text>
-      <MarkdownEditor value={description} maxLength={5000}  />
-
-      <Form.Item
-        label={t("project.change.visible")}
-        required
-        name="visible"
-        valuePropName="checked"
-      >
-        <Switch />
-      </Form.Item>
-      <Form.Item
-        label={t("project.change.maxScore")}
-        name="maxScore"
-        tooltip={t("project.change.maxScoreHelp")}
-        rules={[{ required: false, message: t("project.change.maxScoreMessage") }]}
-      >
-        <Input
-          min={1}
-          max={1000}
-          type="number"
-        />
-      </Form.Item>
-      <Form.Item
-        label={t("project.change.deadline")}
-        name="deadline"
-        rules={[{ required: true }]}
-      >
-        <DatePicker
-          showTime
-          format="YYYY-MM-DD HH:mm:ss"
-        />
-      </Form.Item>
-    </>
-  )
+    const { t } = useTranslation()
+    const description = Form.useWatch("description", form)
+    const visible = Form.useWatch("visible", form)
+    const [isVisible, setIsVisible] = useState(visible)
+    const [savedVisibleAfter, setSavedVisibleAfter] = useState<string | null>(null)
+
+    useEffect(() => {
+        setIsVisible(visible)
+        if (visible && savedVisibleAfter) {
+            form.setFieldsValue({ visibleAfter: null })
```

Comment:
Ik denk dat het gemakkelijker is om dit te doen net voor je de POST/PUT request maakt.

Reply:
Ik heb deze logica verplaatst naar de handleCreation functies van EditProject en ProjectCreate en de overbodige code uit GeneralFormTab verwijderd